plugin/trino-postgresql/src/test/java/io/trino/plugin/postgresql/TestPostgreSqlConnectorTest.java

```diff
@@ -283,17 +281,10 @@ public void testPredicatePushdown()
                 .matches("VALUES " +
                         "(BIGINT '3', BIGINT '19', CAST('ROMANIA' AS varchar(25))), " +
                         "(BIGINT '2', BIGINT '21', CAST('VIETNAM' AS varchar(25)))")
-                // Filter node is retained as no constraint is pushed into connector.
-                // The compacted domain is a range predicate which can give wrong results
-                // if pushed down as PostgreSQL has different sort ordering for letters from Trino
                 .isNotFullyPushedDown(
                         node(
                                 FilterNode.class,
-                                // verify that no constraint is applied by the connector
-                                tableScan(
-                                        tableHandle -> ((JdbcTableHandle) tableHandle).getConstraint().isAll(),
-                                        TupleDomain.all(),
-                                        ImmutableMap.of())));
+                                PlanMatchPattern.node(TableScanNode.class)));
```

Comment:
why still `isNotFullyPushedDown`?

Reply:
filterPredicate seems to remain on Trino side although the compacted predicate is pushed down to PostgreSQL:
```sql
SELECT "nationkey", "name", "regionkey" FROM "tpch"."nation" WHERE ("name" >= ? COLLATE "C" AND "name" <= ? COLLATE "C")
```
```
Output[regionkey, nationkey, name]
│   Layout: [regionkey:bigint, nationkey:bigint, name:varchar(25)]
└─ RemoteExchange[GATHER]
   │   Layout: [nationkey:bigint, name:varchar(25), regionkey:bigint]
   └─ ScanFilter[table = postgresql:tpch.nation tpch.nation constraint on [name] columns=[nationkey:bigint:int8, name:varchar(25):varchar, regionkey:bigint:int8], filterPredicate = ("name" IN (CAST('POLAND' AS varchar(25)), CAST('ROMANIA' AS varchar(25)), CAST('VIETNAM' AS varchar(25))))]
          Layout: [nationkey:bigint, name:varchar(25), regionkey:bigint]
          nationkey := nationkey:bigint:int8
          regionkey := regionkey:bigint:int8
          name := name:varchar(25):varchar
```